python/TestHarness/schedulers/Scheduler.py

```diff
@@ -292,16 +292,14 @@ def schedule(self, testers):
         # Assign a status thread to begin work on any skipped/failed jobs
         self.queueJobs(status_jobs=non_runnable_jobs)
 
-        # Build our list of runnable jobs and set the tester's status to queued
-        job_list = []
-        if runnable_jobs:
-            job_list = job_dag.ind_nodes()
-            for job_container in job_list:
-                tester = job_container.getTester()
-                tester.setStatus('QUEUED', tester.bucket_pending)
+        # Set original DAG state now, before we launch any jobs
+        for job_container in job_dag.topological_sort():
+            job_container.setOriginalDAG()
```

Comment:
With my first comment above, this is what I am talking about. Having to set the original DAG now that its complete. It must be done for each job object because it is _not_ shared.

Reply:
You have a few options. We are working on the assumption that you build the DAG, and then you use the DAG. We don't build the DAG, use it a little, then add to it some more. If we assume that workflow, then you definitely can remove the explicit call to "set the DAG" (not sure I love that name). You simply set the DAG the very first time you try to use it. To do that, you'd have to modify the underlying DAG class so that when you call topological sort or whatever else on it, that you check some Boolean and call that set method on it before you start any other operation. I'll go through your PR and see if I understand what you are doing.

Reply:
hmmm, perhaps when we perform a  dag.ind_nodes() do get a list of concurrent jobs... when thats called, we should assume the dag is ready to rock...